node/node.go

```diff
@@ -649,6 +649,18 @@ func (node *AlgorandFullNode) GetPendingTransaction(txID transactions.Txid) (res
 		minRound++
 	}
 
+	// If we did find the transaction, we know there is no point
+	// checking rounds earlier or later than validity rounds
+	if found {
+		if tx.Txn.FirstValid > minRound {
+			minRound = tx.Txn.FirstValid
+		}
+
+		if tx.Txn.LastValid < maxRound {
+			maxRound = tx.Txn.LastValid
+		}
```

Comment:
As long as `tx.Txn.FirstValid` is guaranteed to be before `tx.Txn.LastValid` at this point (which I expect it must be, if the txn is allowed in the pending pool), this looks correct to me